extensions/ql-vscode/src/cli.ts

```diff
@@ -614,6 +614,26 @@ export class CodeQLCliServer implements Disposable {
       "Resolving qlpack information",
     );
   }
+
+  /**
+   * Gets information about queries in a query suite.
+   * @param additionalPacks A list of directories to search for qlpacks before searching in `searchPath`.
+   * @param searchPath A list of directories to search for packs not found in `additionalPacks`. If undefined,
+   *   the default CLI search path is used.
+   * @returns A list of query files found
+   */
+  resolveQueriesInSuite(suite: string, additionalPacks: string[], searchPath?: string[]): Promise<string[]> {
+    const args = ['--additional-packs', additionalPacks.join(path.delimiter)];
+    if (searchPath !== undefined) {
+      args.push('--search-path', path.join(...searchPath));
+    }
+    args.push(suite);
+    return this.runJsonCodeQlCliCommand<string[]>(
```

Comment:
Should you mark this function as `async`?

Reply:
It's a fair question, and one which applies to a bunch of other `Promise`-returning functions in this file, but I think that I like the existing convention that we mark a function as `async` if it needs to be, i.e., actually uses `await`, and otherwise let its return type being `Promise` being a sufficient signal that what it returns isn't immediately a value but a future.